src/vehicle.cpp

```diff
@@ -4693,7 +4773,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+            if( chainsaw_on && part_info( ret.part ).has_flag( "CHAINSAW" ) &&
+                parts[part].hp > 0 ) {
+                dam += 66;
```

Comment:
This looks better.
I'd decrease it further by target's armor (so that it effectively had negative armor piercing) to avoid it sawing through metallic enemies and hulks.

Chainsaws should be effective against soft critters, but brutes and hulks should either slam them off or clog them with their flesh.

Though now that I think about it, collision might not be the best place to trigger it. Consider zombies climbing over a box of chainsaws - the chainsaws won't trigger in such a case.